File: StructuralPatterns/Bridge/TypeScript/Circle.ts
import { Shape } from './Shape';
import { Renderer } from './Renderer';

export class Circle extends Shape {
    private readonly radius: number;
    constructor(renderer: Renderer, radius: number) {
        super(renderer);
        this.radius = radius;
    }
    draw(): void {
        this.renderer.renderCircle(this.radius);
    }
}

File: StructuralPatterns/Bridge/Go/circle.go
package main

type Circle struct {
	renderer Renderer
	radius   float64
}

func NewCircle(renderer Renderer, radius float64) *Circle {
	return &Circle{renderer: renderer, radius: radius}
}

func (c *Circle) Draw() {
	c.renderer.RenderCircle(c.radius)
}
